SumoLogic.Logging.NLog/BufferedSumoLogicTarget.cs

```diff
@@ -306,7 +307,11 @@ protected override void InitializeTarget()
                 sourceHost,
                 this.LogLog);
 
-            this.flushBufferTimer = new Timer((s)=> this.flushBufferTask.Run(), null, TimeSpan.FromMilliseconds(0), TimeSpan.FromMilliseconds(this.FlushingAccuracy));
+            this.flushBufferTimer = new Timer(
+                async _ => await flushBufferTask.Run().ConfigureAwait(false),

```

Comment:
Hit a `System.Threading.Tasks.TaskCanceledException: 'The operation was canceled.'` when doing stress test

any thought?

```
System.Threading.Tasks.TaskCanceledException
  HResult=0x8013153B
  Message=The operation was canceled.
  Source=System.Net.Http
  StackTrace:
   at System.Net.Http.HttpConnection.<SendAsyncCore>d__61.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at System.Net.Http.HttpConnectionPool.<SendWithNtConnectionAuthAsync>d__40.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at System.Net.Http.HttpConnectionPool.<SendWithRetryAsync>d__39.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at System.Net.Http.RedirectHandler.<SendAsync>d__4.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at System.Net.Http.HttpClient.<FinishSendAsyncBuffered>d__62.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at SumoLogic.Logging.Common.Sender.SumoLogicMessageSender.<TrySend>d__36.MoveNext() in \\Mac\Home\git\sumologic-net-appenders\SumoLogic.Logging.Common\Sender\SumoLogicMessageSender.cs:line 242
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SumoLogic.Logging.Common.Sender.SumoLogicMessageSender.<Send>d__34.MoveNext() in \\Mac\Home\git\sumologic-net-appenders\SumoLogic.Logging.Common\Sender\SumoLogicMessageSender.cs:line 159
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SumoLogic.Logging.NLog.BufferedSumoLogicTarget.<<InitializeTarget>b__66_0>d.MoveNext() in \\Mac\Home\git\sumologic-net-appenders\SumoLogic.Logging.NLog\BufferedSumoLogicTarget.cs:line 311

Inner Exception 1:
IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request.

Inner Exception 2:
SocketException: The I/O operation has been aborted because of either a thread exit or an application request

```

Reply:
What's your `ConnectTimeout` set to? I don't know how many concurrent messages were in flight, but could you raise it to something fairly high (maybe 5x the current setting) so we can rule out a timeout issue?

I just pushed a change to replace the various `Wait()` calls w/ `GetAwaiter().GetResult()`. That will unwind any `AggregateException` and should be more compatible w/ the existing exception handling codepaths (although there's not any explicit `TimeoutException` handling now).

Reply:
Looks issue is gone after applying the new change

Reply:
Created #89 to resolve this.